File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsCourseGradingAgents.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
data class CourseGradingAgentSummary(
    val id: String,
    val itemId: String,
    val itemKind: String? = null,
    val assignmentTitle: String,
    val assignmentArchived: Boolean = false,
    val status: String,
    val autoGradeNew: Boolean = false,
    val hasWorkflowGraph: Boolean = false,
    val updatedAt: String,
    val reviewCount: Int? = null,
)

@Serializable
data class CourseGradingAgentsListResponse(
    val agents: List<CourseGradingAgentSummary> = emptyList(),
)

@Serializable
data class GraderAgentTemplateSummary(
    val id: String,
    val name: String,
    val isBuiltin: Boolean? = null,
    val updatedAt: String,
)

@Serializable
data class GraderAgentTemplatesListResponse(
    val templates: List<GraderAgentTemplateSummary> = emptyList(),
)

@Serializable
data class GraderAgentConfig(
    val id: String? = null,
    val prompt: String = "",
    val includeAssignmentContent: Boolean = false,
    val includeRubric: Boolean = false,
    val status: String = "draft",
    val autoGradeNew: Boolean? = null,
    val postPolicy: String? = null,
    val confidenceFloor: Double? = null,
    val modelId: String? = null,
    val updatedAt: String? = null,
    val workflowGraph: JsonElement? = null,
)

@Serializable
data class GraderAgentConfigEnvelope(
    val config: GraderAgentConfig? = null,
)

@Serializable
data class GraderAgentTemplateDetail(
    val id: String,
    val name: String,
    val prompt: String,
    val includeAssignmentContent: Boolean,
    val includeRubric: Boolean,
    val workflowGraph: JsonElement? = null,
    val createdAt: String,
    val updatedAt: String,
)

@Serializable
data class GraderAgentTemplateDetailEnvelope(
    val template: GraderAgentTemplateDetail,
)

@Serializable
data class PutGraderAgentConfigBody(
    val prompt: String,
    val includeAssignmentContent: Boolean,
    val includeRubric: Boolean,
    val status: String,
    val autoGradeNew: Boolean,
    val postPolicy: String,
    val confidenceFloor: Double? = null,
    val workflowGraph: JsonElement,
)

@Serializable
data class PutGraderAgentConfigResponse(
    val config: GraderAgentConfig,
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsCourseGradingAgents.swift
import Foundation

/// Grading agent settings models (M13.6).
struct CourseGradingAgentSummary: Codable, Identifiable, Hashable {
    var id: String
    var itemId: String
    var itemKind: String?
    var assignmentTitle: String
    var assignmentArchived: Bool
    var status: String
    var autoGradeNew: Bool
    var hasWorkflowGraph: Bool
    var updatedAt: String
    var reviewCount: Int?
}

struct CourseGradingAgentsListResponse: Codable {
    var agents: [CourseGradingAgentSummary]
}

struct GraderAgentTemplateSummary: Codable, Identifiable, Hashable {
    var id: String
    var name: String
    var isBuiltin: Bool?
    var updatedAt: String
}

struct GraderAgentTemplatesListResponse: Codable {
    var templates: [GraderAgentTemplateSummary]
}

struct GraderAgentConfig: Codable, Hashable {
    var id: String?
    var prompt: String
    var includeAssignmentContent: Bool
    var includeRubric: Bool
    var status: String
    var autoGradeNew: Bool?
    var postPolicy: String?
    var confidenceFloor: Double?
    var modelId: String?
    var updatedAt: String?
    var workflowGraph: JSONValue?
}

struct GraderAgentConfigEnvelope: Decodable {
    var config: GraderAgentConfig?
}

struct GraderAgentTemplateDetail: Codable {
    var id: String
    var name: String
    var prompt: String
    var includeAssignmentContent: Bool
    var includeRubric: Bool
    var workflowGraph: JSONValue?
    var createdAt: String
    var updatedAt: String
}

struct GraderAgentTemplateDetailEnvelope: Decodable {
    var template: GraderAgentTemplateDetail
}

struct PutGraderAgentConfigBody: Encodable {
    var prompt: String
    var includeAssignmentContent: Bool
    var includeRubric: Bool
    var status: String
    var autoGradeNew: Bool
    var postPolicy: String
    var confidenceFloor: Double?
    var workflowGraph: JSONValue
}

struct PutGraderAgentConfigResponse: Decodable {
    var config: GraderAgentConfig
}
